### ISSUE TITLE
Assign mulitple floating IPs to different nodes

### ISSUE THREAD

**Issue by cbeneke** [cbeneke/hcloud-fip-controller#32]
Currently the leader attaches all floating IPs to it's own node. To reduce load and impact on single node failure it should attach each floating IP to a different node (e.g. dedicated routine incl. leader election for each floating IP)

**Comment by vitobotta** [comment]
Hi @cbeneke! Is this not possible by installing one controller per IP? At the moment I am assigning three IPs manually, one per each of my three nodes, so I can distribute load for nginx controllers. If I install one controller per IP, is there a way at all now to ensure each IP is assigned to a different node? Thanks!

**Comment by cbeneke** [comment]
Hi, yes this is the only way to do it currently. But I'd like to build it in a way, that you don't need as many pods (also when you install one deployment per IP and have them scheduled to the same nodes the IPs will still be attached to the same nodes).

If you have many IPs it might be a problem to install one controller per IP :)

**Comment by vitobotta** [comment]
Hi! I was just checking the helm chart repo, and in the values.yml file I can see that I can specify a node selector. So if I install multiple controllers with different node selectors it should work, right? Or is there any reason why it wouldn't? I am using Rancher with [a node driver](https://github.com/mxschmitt/ui-driver-hetzner) to manage node pools, so when I change the node pools I would like that the IPs are reassigned to the new nodes automatically while keeping the one per node distribution. It's OK if I have to label the nodes. Can you give me some more details, if you don't mind, about the anti affinity option that you suggest instead? Thanks!

**Comment by vitobotta** [comment]
I see the `affinity` setting in the values.yml but not sure of how to use it. :)

**Comment by vitobotta** [comment]
Got it working with pod antiaffinity - the IPs were assigned to different nodes. Yay! I just had a couple of issues with the helm chart described in https://github.com/cbeneke/helm-charts/issues/4. I will try to submit a PR.

**Comment by cbeneke** [comment]
To answer your questions: The controller does not schedule the IPs to a random node, but always the node it is running on. Therefor each kubernetes scheduling policy (e.g. nodeSelectors) should directly work when applied to the fip-controller pods.

**Comment by vitobotta** [comment]
Uhm.... I spoke too soon. It doesn't always run the pods on different nodes. This is the anti affinity config I am using:

```yaml
affinity:
  podAntiAffinity:
    requiredDuringSchedulingIgnoredDuringExecution:
      - labelSelector:
          matchExpressions:
            - key: "app.kubernetes.io/name"
              operator: In
              values:
                - hcloud-fip-controller
        topologyKey: "kubernetes.io/hostname"
```

What am I doing wrong?

**Comment by vitobotta** [comment]
I am specifying the namespaces now and it seems to work better.

**Comment by cbeneke** [comment]
Afaict the anti-affinity rule looks good, but if the deployments are running in different namespaces you have to ensure to list them, because the pod affinity rules are 'namespaced' (as pods are namespaced) .. compare https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#inter-pod-affinity-and-anti-affinity ^^

**Comment by vitobotta** [comment]
Do you know if it's possible to tell the antiaffinity to look in all namespaces so we don't have to specify manually? I am looking to automate this with Pulumi.

**Comment by cbeneke** [comment]
There is an open issue for that (https://github.com/kubernetes/kubernetes/issues/68827).

**Comment by vitobotta** [comment]
Nice :)

**Comment by phiphi282** [comment]
Will be implemented in v0.4.0 (see #39 )
